core/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java

```diff
@@ -1507,7 +1507,7 @@ private void rewriteMerge(SqlMerge call) {
     // note that the values clause has already been converted to a
     // select on the values row constructor; so we need to extract
     // that via the from clause on the select
-    if (insertCall != null) {
+    if (insertCall != null && config.sqlConformance().allowMergeInsertRewrite()) {
```

Comment:
nit: line wrap

Reply:
Fix in https://github.com/googleinterns/calcite/pull/139/commits/20a224efcc552eb936868ade31f0f9883ea844f6